testnet/src/tests/mod.rs

```diff
@@ -505,3 +506,30 @@ fn should_succeed_handling_malformed_and_valid_txs() {
     });
     run_loop.start(num_rounds);
 }
+
+#[test]
+fn test_btc_to_sat() {
+    let inputs = [
+        "0.1000000",
+        "0.0000001",
+        "1.0",
+        "0.1",
+    ];
+    let expected_outputs: [u64; 4] = [
+        1000000,
```

Comment:
This is incorrect -- 0.1 BTC = 10 million satoshis (you're off by a power of 10).